src/Datadog.Trace/TracingProcessManager.cs

```diff
@@ -15,51 +15,73 @@ namespace Datadog.Trace
 {
     internal class TracingProcessManager
     {
-        private static readonly List<ProcessMetadata> Processes = new List<ProcessMetadata>()
+        private static readonly ProcessMetadata TraceAgentMetadata = new ProcessMetadata()
         {
-            new ProcessMetadata()
+            Name = "datadog-trace-agent",
+            ProcessPathKey = ConfigurationKeys.TraceAgentPath,
+            ProcessArgumentsKey = ConfigurationKeys.TraceAgentArgs,
+            PreStartAction = () =>
             {
-                Name = "datadog-trace-agent",
-                ProcessPathKey = ConfigurationKeys.TraceAgentPath,
-                ProcessArgumentsKey = ConfigurationKeys.TraceAgentArgs,
-                PreStartAction = () =>
+                TraceAgentMetadata.Port = FreeTcpPort();
```

Comment:
I didn't catch this in the previous PR, but can we rename `FreeTcpPort()` to `GetFreeTcpPort()`. Otherwise it sounds like a method that frees up a port.